packages/vite/src/client.ts

```diff
@@ -59,6 +60,13 @@ export async function buildClient (nuxt: Nuxt, ctx: ViteBuildContext) {
 
   await nuxt.callHook('vite:extendConfig', clientConfig, { isClient: true, isServer: false })
 
+  // Add transpile packages to optimizeDeps.exclude to prevent Vite from optimizing them at runtime
+  // This must happen after vite:extendConfig since modules may add transpile entries asynchronously
+  clientConfig.optimizeDeps ||= {}
+  clientConfig.optimizeDeps.exclude ||= []
+  const transpilePatterns = getTranspilePatterns({ isDev: nuxt.options.dev, isClient: true })
```

Comment:
we should probably _only_ inject them here, as otherwise we'll likely have duplicates.